tuxedo-core/aggregator/src/lib.rs

```diff
@@ -115,12 +130,28 @@ pub fn tuxedo_constraint_checker(attrs: TokenStream, body: TokenStream) -> Token
     let variants = variant_type_pairs.clone().map(|(v, _t)| v);
     let inner_types = variant_type_pairs.map(|(_v, t)| t);
 
-    let vis = ast.vis;
+    // Set up the names of the new associated types.
     let mut error_type_name = outer_type.to_string();
     error_type_name.push_str("Error");
     let error_type = Ident::new(&error_type_name, outer_type.span());
-    let inner_types = inner_types.clone();
+
+    let mut inherent_hooks_name = outer_type.to_string();
+    inherent_hooks_name.push_str("InherentHooks");
+    let inherent_hooks = Ident::new(&inherent_hooks_name, outer_type.span());
+
+    let vis = ast.vis;
+
+    // TODO there must be a better way to do this, right?
```

Comment:
The first problem here is that the value can't be cloned in the quote macro (as far as I know, if it would be possible with some magic trick then the solution would be obvious) so all the clones need to be done before the macro, and that repeated code is the result. An easy solution is to involve a macro_rules to avoid repeat all the syntax but the inner problem is that inside macro_rules identifiers can't be created, then I came up with two solutions:
1. the first one need to provide to the macro the name of the variables and the name of the variable that needs to be copied, but this result into more lines and even uglier then the repeated code
2. Using the `paste` macro, that enables you to append identifiers to create new ones, the code is the following:

```rust
    macro multiple_assign {
        ($($num:literal),* = $var:ident) => {
            paste!{ $(let [<$var $num>] = $var.clone();)* }
        }
    }
    
    multiple_assign!(2,3,4,6 = inner_types);
    multiple_assign!(2,3,4,5,6 = variants);
```

the final code is not smaller than the previous one but is more scalable if the required copies of inner_types and variants will grow in the future